thirdparty/faiss/faiss/IndexIVFScalarQuantizerCC.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <faiss/IndexScalarQuantizer.h>
+#include <cstdio>
+#include <fstream>
+#include <optional>
+#include "knowhere/utils.h"
+namespace faiss {
+
+/***************************************************
+ *IndexIVFScalarQuantizerCC
+ ***************************************************/
+struct IndexIVFScalarQuantizerCC : IndexIVFScalarQuantizer {
+    struct DiskBackupHandler {
+       public:
+        DiskBackupHandler(std::string prefix, size_t block_size);
+        ~DiskBackupHandler();
+        void ReadDataBlock(char* data, size_t block_id);
+        void AppendDataBlock(const char* data);
+
+       private:
+        std::shared_mutex file_mtx;
+        size_t buffer_size;
+        size_t buffer_res_size;
+        std::unique_ptr<char[]> buffer;
+        std::string raw_data_file_name;
+        size_t block_size;
+        size_t file_block_num;
+        size_t buffer_block_num;
+        size_t buffer_max_block_num;
```

Comment:
It seems that this module is a common file holder and it can be independent with ScalarQuantizerCC. It could be better to place this class into a standalone class, and other index could also reuse it (like ivfpqcc)

Reply:
updated.